Bug Fix: on MacOS Array.findLast is not implemented, use polyfill

doctorguile opened the pull request:
Thanks for the work! 
I found that on on MacOS 14.2 anki 2.1, I need a polyfill for Array.findLast

doctorguile committed:
on MacOS Array.findLast is not implemented, use polyfill

nihil-admirari commented:
`findLast` is reported as being supported since Safari 15.4: https://caniuse.com/mdn-javascript_builtins_array_findlast, which predates Sonoma by a couple of years.

If you mean desktop Anki, please try the latest Qt 6 version https://github.com/ankitects/anki/releases/tag/23.10.1. It must support `findLast`.

doctorguile committed:
Rename "Back Extra" field to "Extra"
In the file `_cloze-overlapper.js`, a polyfill for `Array.findLast` is added to support Anki 2.1 on MacOS 14.2 Sonoma.

fix a bug where the cloze-setting might have leading trailing whitespace
and we need to trim first

In the file `back.template.anki` and `front.template.anki`, the template field `{{Before|After|OnlyContext|RevealAll|InactiveHints}}` is replaced with `{{Settings}}`.

nihil-admirari commented:
Have you actually tried the latest Anki version? Ancient Anki versions are not supported, only the latest ones are.

Also, the latest push brought in a number of completely unrelated changes.

doctorguile commented:
@nihil-admirari I'm using Version ⁨2.1.61 (0c1eaf4c)⁩, you are probably right that the latest anki might not have this problem.

I'm closing this PR. If someone running an older version of anki like me they can at least have a work around by patching it themselves.

Thanks
